crates/uv-client/src/registry_client.rs

```diff
@@ -78,19 +80,25 @@ impl RegistryClientBuilder {
         self
     }
 
+    #[must_use]
+    pub fn client(mut self, client: Client) -> Self {
+        self.client = Some(client);
+        self
+    }
+
     pub fn build(self) -> RegistryClient {
-        let client_raw = {
+        let client_raw = self.client.unwrap_or_else(|| {
             // Get pip timeout from env var
             let default_timeout = 5 * 60;
             let timeout = env::var("UV_REQUEST_TIMEOUT")
```

Comment:
Perhaps mildly confusing that `UV_REQUEST_TIMEOUT` is ignored here, but not an issue in practice.